common/sgx/eeid.c

```diff
@@ -494,21 +515,26 @@ oe_result_t verify_eeid(
     const uint8_t* reported_enclave_signer,
     uint16_t reported_product_id,
     uint32_t reported_security_version,
-    uint64_t reported_attributes,
+    uint64_t reported_oe_attributes,
+    const sgx_attributes_t* reported_sgx_attributes,
+    uint32_t reported_misc_select,
     const uint8_t** base_enclave_hash,
-    const oe_eeid_t* eeid)
+    const oe_eeid_t* eeid,
+    const oe_enclave_size_settings_t* base_image_sizes)
 {
     oe_result_t result = OE_UNEXPECTED;
 
     if (!eeid)
         OE_RAISE(OE_INVALID_PARAMETER);
 
     if (eeid->signature_size != 1808) // We only support SGX sigstructs for now.
-        OE_RAISE(OE_VERIFY_FAILED);
+        OE_RAISE(OE_UNSUPPORTED);
+
+    bool static_sizes = !is_eeid_base_image(base_image_sizes);
```

Comment:
Instead of using the base_image_sizes provided by the attestating enclave code, can use the `size_settings` in eeid data structure to drive the static vs. dynamic size logic. A "static size" EEID enclave is suppoed to have al-0 `size_setting` in eeid. The eeid data is measured, any spoofing or tampering of eeid data will be caught by mesurement checks.